False-positive `N802` with `overload` and `override`

* A minimal code snippet that reproduces the bug.

```python
from typing import overload

from typing_extensions import override


class CannotModify:
    @overload
    def BadName(self, var: int) -> int:  # noqa: N802
        ...

    @overload
    def BadName(self, var: str) -> str:  # noqa: N802
        ...

    def BadName(self, var: int | str) -> int | str:  # noqa: N802
        return var


class MyClass(CannotModify):
    @overload
    def BadName(self, var: int) -> int:
        ...

    @overload
    def BadName(self, var: str) -> str:
        ...

    @override
    def BadName(self, var: int | str) -> int | str:
        return var * 2

```

* The command you invoked
```
ruff --isolated --select N802 bug.py
```

gives 

```
bug.py:21:9: N802 Function name `BadName` should be lowercase
bug.py:25:9: N802 Function name `BadName` should be lowercase
Found 2 errors.
```

I feel the `@override` before the implementation of `MyClass.BadName` should extend to the overloads. Or should I add one `@override` per `@overload`?

* The current Ruff version (`ruff --version`).

0.0.290

--- a/crates/ruff/resources/test/fixtures/pep8_naming/N802.py
+++ b/crates/ruff/resources/test/fixtures/pep8_naming/N802.py
@@ -41,9 +41,13 @@ class Test(unittest.TestCase):
         assert True
 
 
-from typing import override
+from typing import override, overload
 
 
 @override
+def BAD_FUNC():
+    pass
+
+@overload
 def BAD_FUNC():
     pass

--- a/crates/ruff/src/rules/pep8_naming/rules/invalid_function_name.rs
+++ b/crates/ruff/src/rules/pep8_naming/rules/invalid_function_name.rs
@@ -70,9 +70,12 @@ pub(crate) fn invalid_function_name(
         return None;
     }
 
-    // Ignore any functions that are explicitly `@override`. These are defined elsewhere,
-    // so if they're first-party, we'll flag them at the definition site.
-    if visibility::is_override(decorator_list, semantic) {
+    // Ignore any functions that are explicitly `@override` or `@overload`.
+    // These are defined elsewhere, so if they're first-party,
+    // we'll flag them at the definition site.
+    if visibility::is_override(decorator_list, semantic)
+        || visibility::is_overload(decorator_list, semantic)
+    {
         return None;
     }
 

